src/lib/caching/structures/Message.ts

```diff
@@ -254,6 +254,15 @@ export class Message extends Structure {
 		return new ReactionCollector(this, options).collect();
 	}
 
+	/**
+	 * Crosspost this message.
+	 * @since 0.0.4
+	 */
+	public crosspost(): Promise<this> {
+		if (!(this.channel instanceof NewsChannel)) return Promise.reject(new Error('Messages can only be crossposted in NewsChannels.'));
```

Comment:
`if (this.channel.type !== ChannelTypes.GroupNews) return Promise.reject(...);` would be more ideal. This would work because the channel type is discriminative, `Channels#type` being `ChannelTypes.GroupNews` is only true for the `NewsChannel`.

Reply:
And while we're at it, you can make `Message#channel` use the `TextBasedChannels` type from this repo's utils.